hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hss/HssCallAttempt.java

```diff
@@ -149,4 +155,47 @@ public boolean isScheduleRedirect() {
         }
         return null;
     }
+
+    /**
+     * Extracts the key set for scheduled calls.
+     *
+     * @return the key set
+     */
+    public Set<Key> keySetFor() {
+        final var sender = nativeOperations().getAccount(senderId());
+        requireNonNull(sender);
+        if (sender.smartContract()) {
+            return getKeysForContractSender();
+        } else {
+            return getKeysForEOASender();
+        }
+    }
+
+    @NonNull
+    private Set<Key> getKeysForEOASender() {
+        // If an Eth sender key is present, use it. Otherwise, use the account key if present.
+        Key key = enhancement.systemOperations().maybeEthSenderKey();
+        if (key != null) {
+            return Set.of(key);
+        }
+        return nativeOperations().authorizingSimpleKeys();
+    }
+
+    @NonNull
+    public Set<Key> getKeysForContractSender() {
+        final var contractNum = maybeMissingNumberOf(senderAddress(), nativeOperations());
+        if (contractNum == MISSING_ENTITY_NUMBER) {
```

Comment:
⚠️ How is this is a possible valid outcome? This could be hiding a bug if it is necessary.